### PR TITLE
Add org.mintguide.WebAppReader

### PR DESCRIPTION
**Pull request by ShvonderSiN**
WebAppReader is an application for  viewing and downloading websites offline on different platforms.

### Please confirm your submission meets all the criteria

- [X ] Please describe your application briefly.
**WebAppReader** is an application for viewing and downloading websites offline
- [X ] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- **filesystem=home**  # to be able to save html pages to a hard drive
- **hare=network** # it is necessary to download html from the Internet
- **вevice=dri** # For the Qt webengine module to work perfectly with gpu
- **env=QTWEBENGINEPROCESS_PATH=/app/bin/QtWebEngineProcess** # necessary for QtWebEngineProcess to work
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### Notes for reviewers
For the application to work correctly, the wget utility version 2.1 is required, since in the older version in upstream 2.0 there are problems with download options, some do not work correctly. The required version is included in the build to ensure stable operation of the application.

### CODE REVIEW

File: org.mintguide.WebAppReader.yaml [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+      - install -Dm644 webappreader.py -t /app/webappreader
+    sources:
+      - type: git
+        url: https://github.com/ShvonderSiN/WebAppReader.git
+        commit: 8577a848c3032e298c5f44444221a294ca19f25c
+      - type: script
+        commands:
+          - 'python3 /app/webappreader/webappreader.py "$@"'
+        dest-filename: webappreader.sh
+    post-install:
```
Comment (hfiguiere) [inline org.mintguide.WebAppReader.yaml:50]: As said in the PR you closed:

put the `post-install` comands into the `build-commands`.

File: org.mintguide.WebAppReader.yaml [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+      - install -Dm644 src/logo_256.png /app/share/icons/hicolor/256x256/apps/org.mintguide.WebAppReader.png
+      - install -Dm755 webappreader.sh /app/bin/webappreader.sh
+      - install -Dm644 flatpak/org.mintguide.WebAppReader.desktop /app/share/applications/org.mintguide.WebAppReader.desktop
+      - install -Dm644 flatpak/org.mintguide.WebAppReader.metainfo.xml /app/share/metainfo/org.mintguide.WebAppReader.metainfo.xml
+      
+  # depencies
+  - {"name": "python3-requirements",
```
Comment (hfiguiere) [inline org.mintguide.WebAppReader.yaml:57]: JSON in YAML?

Comment (bbhtt) [reply org.mintguide.WebAppReader.yaml:57]: This entire part should be in a separate json file, as it is autogenerated by flatpak-pip-generator.

Comment (ShvonderSiN) [reply org.mintguide.WebAppReader.yaml:57]: I just wanted the build to be, just one manifest file, so I put everything in the manifest.. If that's not allowed, I'll do as you say...Should I put json dependencies in this pull request next to the manifest?

File: org.mintguide.WebAppReader.yaml (hunk 1 above)
Comment (ShvonderSiN) [reply org.mintguide.WebAppReader.yaml]: done

File: org.mintguide.WebAppReader.yaml (hunk 2 above)
Comment (ShvonderSiN) [reply org.mintguide.WebAppReader.yaml]: so I made the json file an additional module and put it next to the manifest

**Comment by ShvonderSiN** [comment]
Changes have been made, please re-consider

**Comment by bbhtt** [comment]
bot, build org.mintguide.WebAppReader

**Comment by bbhtt** [comment]
Crashes on clicking download

```
Traceback (most recent call last):
  File "/app/webappreader/pages/download_page.py", line 292, in __start_download
    self.activeDownloadThreads.append(self.download_thread)
                                      ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'DownloadPage' object has no attribute 'download_thread'
/app/bin/webappreader.sh: line 2:     3 Aborted                 (core dumped) python3 /app/webappreader/webappreader.py "$@"
```

**Comment by ShvonderSiN** [comment]
Did you test this? may I ask which site you specified to download?

**Comment by ShvonderSiN** [comment]
Changes have been made, please re-consider.
I put the patch on, can I test it again?

**Comment by bbhtt** [comment]
bot, build org.mintguide.WebAppReader

**Comment by bbhtt** [comment]
How does this work? I can't get it to download https://en.m.wikipedia.org/wiki/Advanced_Video_Coding this page.

[Screencast from 2024-03-25 10-45-18.webm](https://github.com/flathub/flathub/assets/62639087/9645a48d-3ddf-44ba-bd21-c4672fd9b917)

**Comment by ShvonderSiN** [comment]
Some sites do not want to be copied from them and they are protected, I have not done any workarounds to bypass protection. In this case, you can save the page from the browser and add it to the program. Basically, portals where documentation and other manuals work normally... It is very useful.

Lets try somthing like this 
https://faker.readthedocs.io/en/master/
https://docs.djangoproject.com/en/5.0/

https://github.com/flathub/flathub/assets/7835607/4a428a60-5d57-43e2-8cb4-d2be98f0c22d

In any case, I made another commit, where I just changed the message that the site rejected the connection

**Comment by bbhtt** [comment]
bot, build org.mintguide.WebAppReader

**Comment by ShvonderSiN** [comment]
what should i do next?

**Pull request closed without merging**